lib/widgets/search_field.dart

```diff
@@ -0,0 +1,63 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+
+class SearchField extends StatefulWidget {
+  final String labelText;
+  final Color accentColor;
+  final Function(String) onSubmitted;
+
+  const SearchField({
+    required this.labelText,
+    required this.accentColor,
+    required this.onSubmitted,
+  });
+
+  @override
+  _SearchFieldState createState() => _SearchFieldState();
+}
+
+class _SearchFieldState extends State<SearchField> {
+  final FocusNode _focusNode = FocusNode();
+  var _currentColor = Colors.white;
```

Comment:
Instead of storing currentColor, store `_isFocused` and render the color conditionally